ff/src/fields/arithmetic.rs

```diff
@@ -26,7 +26,6 @@ macro_rules! impl_field_mul_assign {
                 {
                     // Tentatively avoid using assembly for `$limbs == 1`.
                     if $limbs <= 6 && $limbs > 1 {
-                        assert!($limbs <= 6);
```

Comment:
should these just be `debug_assert`

Reply:
Well, it doesn't really achieve anything either way since the code will never execute and the assert will never fail.

I think pratyush add these things sometimes hoping that it would help rid of bounds checks that rustc introduces when indexing into slices. But, I have a feeling that indexing into arrays these bounds checks don't happen. 

Further still, in our case we are implementing assembly, so no such bounds checks exist by default.

So they should be removed.

Reply:
Alot of users do compile without assembly, since its inconvenient to do so. This isn't getting called in the assembly code path right?

Reply:
Edit: Sorry you're right this should definitely be removed. Didn't see the use_asm tag right above it. Great catch